R/importSNV.TCGA.R

```diff
@@ -0,0 +1,323 @@
+            myVtypes <- unique(var_type_)
+            if (! all(myVtypes %in% typeVknown)) {
+                # drop file
+                typeValid <- FALSE
+            } else {
+                typeValid <- TRUE
+            }
+
+            # check chromosome value is valid
+            typeChr <- c("chr1","chr2","chr3","chr4","chr5","chr6","chr7","chr8","chr9","chr10","chr11","chr12","chr13",
+                         "chr14","chr15","chr16","chr17","chr18","chr19","chr20","chr21","chr22","chrX","chrY","chrM")
+            myVchr <- unique(chr_)
+            if (! all(myVchr %in% typeChr)) {
+                # drop file
+                chrValid <- FALSE
+            } else {
+                chrValid <- TRUE
+            }
+
+            if (classValid & typeValid & chrValid == FALSE) {
+                # there was an invalid value
+                # report error in file and log
+                # drop file
+                # go to next
+                myNotes <- c(paste(i, " - File was dropped due to invalid attributes in column chr/class/type"))
+                numFilesInvalid <- numFilesInvalid + 1
+            } else {
+                # all tests passed
+                # add data to rSNV
+                fileasDT <- cbind(hugo_ , chr_ , start_, end_, strand_, var_class_,
+                                  var_type_, ref_allele_, tum_all1_, tum_all2_,
+                                  tum_samp_barcode_, tum_sam_uuid_)
+                if (na.rm) {
+                    na.omit(fileasDT)
+                }
+                write.table(fileasDT,  file=rSNV, sep="\t", append=TRUE, row.names =
+                                FALSE, col.names = FALSE, quote = FALSE)
+            }
+            filesFinished <- filesFinished + 1
+        }
+
+
+        # ==== WRITE LOG ===========================================================
+
+        if(writeLog) {
+
+            myTitle <- "importSNV.TCGA"
+
+            # Compile function call record
+            myCall <- character()
+            myCall[1] <- "importSNV.TCGA("
+            myCall[2] <- sprintf("fMAF = \"%s\", ", paste(fMAF, collapse=" "))
+            myCall[3] <- sprintf("rSNV = \"%s\", ", rSNV)
+            myCall[3] <- sprintf("silent = %s, ", as.character(silent))
+            myCall[4] <- sprintf("writeLog = %s)", as.character(writeLog))
+            myCall[5] <- sprintf("na.rm = %s)", as.character(na.rm))
+            myCall <- paste0(myCall, collapse = "")
+
+            # indicate input object name(s)
+            #   (NA for this importSNV.TCGA())
+
+            # Record progress information
+            myNotes <- c(myNotes, sprintf("Files processed - ", filesFinished))
+            myNotes <- c(myNotes, sprintf("Total number of files in input fMAF - ", length(fMAF)))
+            myNotes <- c(myNotes, sprintf("Files dropped due to incomptibility - "), numFilesInvalid)
+
+            # indicate output object name(s)
+            myOutput = c("rSNV")
```

Comment:
That's not an output for which we can process metadata, UUID etc. In the way that we have designed the log-system, this event doesn't actually *have* input and output, because these are files that can be modified on disk. We would only attach metadata to R objects, eg. if we save this as an RDS. Which we don't, in this case.